src/main/java/org/gridsuite/gateway/filters/TokenValidatorGlobalPreFilter.java

```diff
@@ -42,14 +43,18 @@
 public class TokenValidatorGlobalPreFilter extends AbstractGlobalPreFilter {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(TokenValidatorGlobalPreFilter.class);
+    public static final String UNAUTHORIZED_THE_TOKEN_CANNOT_BE_TRUSTED = "{}: 401 Unauthorized, The token cannot be trusted : {}";
 
     private final GatewayService gatewayService;
 
+    private final CacheService cacheService;
+
     @Value("${allowed-issuers}")
     private List<String> allowedIssuers;
 
-    public TokenValidatorGlobalPreFilter(GatewayService gatewayService) {
+    public TokenValidatorGlobalPreFilter(GatewayService gatewayService, CacheService cacheService) {
         this.gatewayService = gatewayService;
+        this.cacheService = cacheService;
```

Comment:
When do we cache the data (public key) ?

Reply:
when we call the getJwksUrl method